tests/tests/rpc_module.rs

```diff
@@ -502,3 +495,36 @@ async fn bounded_subscription_works() {
 		assert_eq!(item, exp);
 	}
 }
+
+#[tokio::test]
+async fn serialize_sub_error_adds_extra_string_quotes() {
```

Comment:
@jsdw @lexnv 

This is a test to illustrate that `StringError` will be serialized with extra string quotes such as `"error":"{...}" even if the value is JSON

We could actually parse the string before adding the `"<string>"` but I came to the solution that is too opinionated and likely better for folks to implement their own type then.

Reply:
Yep, that does make sense! In the end, we will be wrapping the message (may it be a string or already a valid json) with "{ msg }". I'm guessing this is not the common behavior and we don't need to provide a sort of wrapped error to ensure we don't add the extra quotes.
I do wonder, do we have in substrate subscriptions that return directly a valid json for the error? I'd expect that to be unlikely